components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -364,20 +364,21 @@ private void preSetUserClaimsOnEmailUpdate(Map<String, String> claims, UserStore
         if (StringUtils.isNotBlank(emailAddress)) {
 
             String existingEmail;
-            String username = IdentityUtil.addDomainToName(user.getUserName(), user.getUserStoreDomain());
+            String username = user.getUserName();
             try {
                 existingEmail = userStoreManager.getUserClaimValue(username,
                         IdentityRecoveryConstants.EMAIL_ADDRESS_CLAIM, null);
             } catch (UserStoreException e) {
-                throw new IdentityEventException("Error occurred while retrieving existing email address for user: "
-                        + username, e);
+                String error = String.format("Error occurred while retrieving existing email address for user: %s in " +
+                        "domain : %s", user, user.getTenantDomain());
```

Comment:
user will be an object right, we need to print the username here.

Reply:
Fixed: 105be10d6a8db9038911f8f4bdaf0f16e32fae94